### DISCUSSION TITLE
Treat buttons independently?

### DISCUSSION THREAD

**Discussion by Mango0x45** [elkowar/eww#890] [Q&A] [answered]
I’m not quite sure how best to title this question, but basically I have a button for each of my workspaces in my window manager. Each of these buttons has a padding applied to them, and they all look quite uniform. The moment though that a workspace name is a bit longer than the others, instead of growing itself, _all_ the other buttons grow.

Here’s an example to make what I’m saying clearer:

![image](https://github.com/elkowar/eww/assets/57815710/d2e165f4-4f50-4559-87a8-62d7d6d865ec)

![image](https://github.com/elkowar/eww/assets/57815710/17eaa9b9-a58d-4cb6-884e-d63aa46d12ca)

The top bar is waybar, and what I am trying to achieve. The bottom bar is eww. My styling and yuck are all very simple, you can find those [here](https://paste.thomasvoss.com/331.scss) and [here](https://paste.thomasvoss.com/332.yuck)

**Comment by w-lfchen** [comment]
The second link is dead, so I can't view your yuck, therefore I'll have to take a guess.
Try setting the `:space-evenly`-property of the box containing the buttons to `false`, that might be what is causing this.
By default, it's set to `true`.

**Reply by Mango0x45** [reply]
Oh sorry, here's the correct second link: https://paste.thomasvoss.com/332.el

**Reply by w-lfchen** [reply, marked as answer]
![image](https://github.com/elkowar/eww/assets/115360611/812287d5-611e-4173-9efc-907c86473e4f)
```
...
(box
  :class "workspaces"
  :halign "start"
  :space-evenly false
  (for id in workspaces
...
```
Top bar is yours without modification, the bottom one uses an altered workspaces widget with an added `:space-evenly false` line.
I think that might be what you're looking for.

**Reply by Mango0x45** [reply]
Yeah that was it, thank you
